src/main/java/me/arminb/sara/dao/QuestionDAOImpl.java

```diff
@@ -0,0 +1,181 @@
+package me.arminb.sara.dao;
+
+import com.mongodb.*;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.FindOneAndUpdateOptions;
+import com.mongodb.client.model.UpdateOptions;
+import com.mongodb.client.result.DeleteResult;
+import com.mongodb.client.result.UpdateResult;
+import com.sun.javadoc.Doc;
+import me.arminb.sara.entities.Answer;
+import me.arminb.sara.entities.Question;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+import static me.arminb.sara.Constants.*;
+
+@Repository("questionDAO")
+public class QuestionDAOImpl implements QuestionDAO {
+
+    Logger logger = LoggerFactory.getLogger(QuestionDAOImpl.class);
+
+    @Autowired
+    private MongoDatabase database;
+
+
+    private Question mapQuestion(Document questionDoc){
+
+        Question question = new Question();
+
+        DateFormat format = new SimpleDateFormat("EEE MMM d HH:mm:ss z yyyy");
+        Date createDate = null;
+        Date modifiedDate = null;
+        try {
+            if (questionDoc.get("created_date") != null){
+                createDate = format.parse(questionDoc.get("created_date").toString());
+            }
+            if (questionDoc.get("modified_date") != null) {
+                modifiedDate = format.parse(questionDoc.get("modified_date").toString());
+            }
+        } catch (ParseException e) {
+            e.printStackTrace();
```

Comment:
never use `e.printStackTrace();`. If you can ignore the exception just log it as WARN and move on. Otherwise do something about it. Here modifiedAt should be set to null. However, because it's been set to null at the beginning you can just do a warn log and move on.